public/docs/ts/latest/cookbook/a11y.jade

```diff
@@ -0,0 +1,1087 @@
+    do we make any promises about the production readiness of our example. Our focus remains on accessibility and how
+    we have the tools to make the most stubbornly inaccessible component accessible.
+
+:marked
+  There are rules governing which keyboard events should ideally be implemented per widget.
+  For example, a regardless of how it is accessed, a `button` should accept focus, react to the
+  mouse `click` event and react to the keyboard `enter` and `space` events.
+
+  *You can read about these [Common Widget Design Patterns](https://www.w3
+  .org/WAI/intro/accessibility.php) at the `W3C`.*
+
+  Let's take a look at our custom button element:
+
++makeTabs('cb-a11y/ts/app/shared/a11y-custom-button.component.ts,cb-a11y/ts/app/shared/a11y-custom-button.component.html',
+null, 'a11y-custom-button.component.ts,a11y-custom-button.component.html')
+
+:marked
+  We manipulate the `Host` element of our component and now it can now be used like the standard
+  `button` element:
+
++makeExample('cb-a11y/ts/app/managing-focus/a11y-managing-focus.component.html','cb-a11y-custom-button-usage')
+
+:marked
+  Looking at the generated HTML we see:
+
+code-example(language="html" escape="html" format="linenums").
+  <a11y-custom-button class="btn btn-primary" role="button" tabindex="0">
+    Do something...
+  </a11y-custom-button>
+
+:marked
+  **Important**:  Note the `role` and `tabindex`. 
+  
+  The `ARIA role` of this element is `button`. It tells any assistive technologies that this element is
+  a button, regardless of the original design of the HTML element. More information on this later.
+  
+  Setting the `tabindex` to `0` inserts the element in the default flow of`keyboard navigation` focus. This means that our
+  element also becomes accessible via the keyboard!
+
+.l-sub-section
+  :marked
+    **DO NOT** use a `tabindex` value of `1` or greater as this will change the default keyboard navigation.
+
+:marked
+  ### Internal focus management for components
+
+  Programmatically setting focus within an app can introduce accessibility issues by interrupting
+  or enforcing a particular flow. If you do set focus for the user, you must ensure that
+  you do so in an accessible way.
+  
+  We'll illustrate how to do this by creating a `button` that shows an `alert`, then sets focus on the `alert`
+  and then allows the user to close the error message with a `close button`.
+
++makeTabs('cb-a11y/ts/app/managing-focus/a11y-error-demo.component.html,cb-a11y/ts/app/managing-focus/a11y-error-demo.component.ts',
+null, 'a11y-error-demo.component.html,a11y-error-demo.component.ts')
+
+:marked
+  We are setting focus on an `alert` that starts out hidden. As `hidden` and `disabled` elements cannot accept focus,
+  we first need this element to become visible again. To give the browser time to apply this change we set our focus using
```

Comment:
Is there a better way to handle this than the timeout? I've typically done the same thing in my angular1 apps but it's always felt like a hack.